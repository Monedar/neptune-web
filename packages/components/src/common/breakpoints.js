export const Breakpoints = [0, 576, 768, 992, 1200];
