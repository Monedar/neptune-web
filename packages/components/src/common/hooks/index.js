export { default as useHasIntersected } from './useHasIntersected';
export { default as usePrevious } from './usePrevious';
export { default as useClientWidth } from './useClientWidth';
