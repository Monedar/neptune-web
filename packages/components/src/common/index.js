export { default as HistoryNavigator } from './historyNavigator';
export { Sizes, SizesSML, SizesSL } from './propsValues/sizes';
export { DateMode } from './propsValues/dateMode';
export { MonthFormat } from './propsValues/monthFormat';
export { Breakpoints } from './breakpoints';
export { Placement } from './placement';
export { addModalOpenBodyClass, removeModalOpenBodyClass } from './DOMOperations';
export { JustifyContent, AlignItems, FlexDirection } from './flexProperties';
export { Theme } from './theme';
