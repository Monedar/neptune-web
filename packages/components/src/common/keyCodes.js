export default {
  DOWN: 40,
  UP: 38,
  ENTER: 13,
  ESCAPE: 27,
  TAB: 9,
  SPACE: 32,
};
