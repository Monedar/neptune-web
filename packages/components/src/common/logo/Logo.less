.np-logo {
  &.flag {
    width: var(--size-24);
    height: var(--size-32);
  }
}
