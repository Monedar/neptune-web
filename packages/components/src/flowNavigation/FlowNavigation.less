@import (reference) '@transferwise/neptune-css/src/less/addons/_spacing-utilities.less';

.np-flow-navigation {
  .p-b-1;

  @media (--screen-sm) {
    .p-b-5;
  }

  &__stepper {
    .p-b-0;
    .p-x-1;
    @media (--screen-sm) {
      .p-x-5;
    }
  }

  .separator {
    .m-x-2;
    height: var(--size-32);
    width: 1px;
    background-color: var(--color-border-neutral);
  }
}
