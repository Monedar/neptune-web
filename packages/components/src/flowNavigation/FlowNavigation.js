import React, { useRef } from 'react';
import Types from 'prop-types';
import classNames from 'classnames';
import Header from '../header';
import Stepper from '../stepper';
import AnimatedLabel from './animatedLabel';
import BackButton from './backButton';
import CloseButton from '../common/closeButton';
import Logo from '../common/logo';

import { Theme, Breakpoint } from '../common';
import { usePrevious, useClientWidth } from '../common/hooks';

import './FlowNavigation.css';

const FlowNavigation = ({ avatar, stepper: { steps, activeStep }, onClose, onGoBack }) => {
  const prev = usePrevious(activeStep);
  const ref = useRef(null);

  const [clientWidth] = useClientWidth({ ref });
  const closeButton = onClose && <CloseButton onClick={onClose} />;

  console.log('clientWidth', clientWidth);

  const getMobileLeftContent = () =>
    onGoBack && activeStep ? (
      <BackButton
        onClick={onGoBack}
        label={
          <AnimatedLabel
            labels={steps.map((step) => step.label)}
            activeStep={activeStep}
            backward={activeStep < prev}
          />
        }
        className="visible-xs"
      />
    ) : (
      <Logo type={Logo.Type.FLAG} className="visible-xs" />
    );

  return (
    <Header
      ref={ref}
      className="np-flow-navigation"
      leftContent={
        <>
          <Logo type={Logo.Type.FULL} className="hidden-xs" />
          {getMobileLeftContent()}
        </>
      }
      rightContent={
        <>
          {avatar}
          {avatar && closeButton && <span className="separator" />}
          {closeButton}
        </>
      }
      bottomContent={
        <Stepper
          activeStep={activeStep}
          steps={steps}
          className={classNames('np-flow-navigation__stepper m-t-1')}
        />
      }
      layout={clientWidth < Breakpoint.MEDIUM ? 'vertical' : 'horizontal'}
    />
  );
};

FlowNavigation.Theme = Theme;

FlowNavigation.defaultProps = {
  avatar: null,
  onGoBack: null,
  onClose: null,
};

FlowNavigation.propTypes = {
  /** The user or business avatar */
  avatar: Types.node,
  /** Function called when the close is clicked */
  onClose: Types.func,
  /** Function called on mobile view when the backButton is clicked */
  onGoBack: Types.func,
  /** Contains the steps to be displayed and the current active Steps. If stepper is not needed please use OverlayHeader */
  stepper: Types.shape({
    activeStep: Types.number,
    steps: Types.arrayOf(
      Types.shape({
        label: Types.node.isRequired,
        onClick: Types.func,
        hoverLabel: Types.node,
      }),
    ),
  }).isRequired,
};

export default FlowNavigation;
