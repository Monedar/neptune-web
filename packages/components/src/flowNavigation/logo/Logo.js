import React from 'react';
import Types from 'prop-types';
import classNames from 'classnames';
import { Theme } from '../../common';
import './Logo.css';

const Logo = ({ className, theme, type }) => {
  let baseClass = ['logo', 'logo-3', 'logo-primary'];
  if (type === Logo.Type.FLAG) {
    baseClass = ['flag', 'flag-info'];
  } else if (theme === Logo.Theme.DARK) {
    baseClass.splice(2, 1, 'logo-inverse');
  }
  return <div className={classNames('np-logo', baseClass, className)} />;
};

Logo.Theme = Theme;

Logo.Type = { FULL: 'FULL', FLAG: 'FLAG' };

Logo.propTypes = {
  className: Types.string,
  theme: Types.oneOf([Logo.Theme.LIGHT, Logo.Theme.DARK]),
  type: Types.oneOf([Logo.Type.FULL, Logo.Type.FLAG]),
};

Logo.defaultProps = {
  className: undefined,
  theme: Theme.LIGHT,
  type: Logo.Type.FULL,
};

export default Logo;
