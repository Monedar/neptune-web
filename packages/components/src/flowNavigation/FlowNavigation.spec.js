import React from 'react';
import '@testing-library/jest-dom';
import { render } from '@testing-library/react';
import { screen } from '@testing-library/dom';

import FlowNavigation from '.';
import { Breakpoint } from '../common';
import Logo from '../common/logo';

jest.mock('lodash.throttle', () => jest.fn((fn) => fn));

jest.mock('./AnimatedLabel', () => {
  // eslint-disable-next-line react/prop-types
  return ({ className }) => <div className={className}>AnimatedLabel</div>;
});
jest.mock('./backButton', () => {
  // eslint-disable-next-line react/prop-types
  return ({ className, label }) => <div className={className}>BackButton{label}</div>;
});
jest.mock('../common/logo', () => {
  // eslint-disable-next-line react/prop-types
  const mockedLogo = ({ className, type }) => <div className={className}>Logo {type}</div>;
  mockedLogo.Type = { FULL: 'FULL', FLAG: 'FLAG' };
  return mockedLogo;
});

describe('FlowNavigation', () => {
  const originalClientWidth = Object.getOwnPropertyDescriptor(HTMLElement.prototype, 'clientWidth');
  const resetClientWidth = (width) => {
    Object.defineProperty(HTMLElement.prototype, 'clientWidth', {
      configurable: true,
      value: width,
    });
  };
  beforeEach(() => {
    resetClientWidth(Breakpoint.MEDIUM + 1);
  });

  afterAll(() => {
    Object.defineProperty(HTMLElement.prototype, 'clientWidth', originalClientWidth);
  });

  const props = {
    avatar: <div />,
    stepper: {
      steps: [
        {
          label: 'label-0',
        },
        { label: 'label-1' },
      ],
      activeStep: 0,
    },
  };
  it(`renders as expected`, () => {
    expect(render(<FlowNavigation {...props} />).container).toMatchSnapshot();
  });

  it(`renders separator only if avatar and onClose are provided `, () => {
    const { container } = render(<FlowNavigation {...props} onClose={jest.fn()} />);
    expect(container.querySelector('.separator')).toBeInTheDocument();
  });

  it(`doesn't render separator if avatar or onClose are not provided `, () => {
    const { container, rerender } = render(<FlowNavigation {...props} />);
    expect(container.querySelector('.separator')).not.toBeInTheDocument();
    rerender(<FlowNavigation {...props} avatar={null} />);
    expect(container.querySelector('.separator')).not.toBeInTheDocument();
  });
  describe('right for mobile', () => {
    beforeAll(() => {});
    it(`renders Logo ${Logo.Type.FLAG} if onGoBack is not provided and activeStep = 0`, () => {
      const { rerender } = render(<FlowNavigation {...props} />);

      expect(screen.getByText(`Logo ${Logo.Type.FLAG}`).parentElement).toHaveClass('visible-xs');

      rerender(
        <FlowNavigation
          {...props}
          stepper={{
            ...props.stepper,
            activeStep: 1,
          }}
        />,
      );

      expect(screen.getByText(`Logo ${Logo.Type.FLAG}`).parentElement).toHaveClass('visible-xs');

      rerender(
        <FlowNavigation
          {...props}
          onGoBack={jest.fn()}
          stepper={{
            ...props.stepper,
            activeStep: 1,
          }}
        />,
      );

      expect(screen.queryByText(`Logo ${Logo.Type.FLAG}`)).not.toBeInTheDocument();
    });

    it(`renders BackButton with AnimatedLabel if onGoBack is provided and activeStep > 0`, () => {
      const { rerender } = render(<FlowNavigation {...props} onGoBack={jest.fn()} />);
      expect(screen.queryByText('BackButton')).toBeNull();

      rerender(
        <FlowNavigation
          {...props}
          stepper={{
            ...props.stepper,
            activeStep: 1,
          }}
        />,
      );
      expect(screen.queryByText('BackButton')).toBeNull();

      rerender(
        <FlowNavigation
          {...props}
          onGoBack={jest.fn()}
          stepper={{
            ...props.stepper,
            activeStep: 1,
          }}
        />,
      );

      expect(screen.getByText('BackButton').parentElement).toHaveClass('visible-xs');
      expect(screen.getByText('AnimatedLabel')).toBeInTheDocument();
    });
  });
  describe(`on mobile`, () => {
    beforeEach(() => {
      resetClientWidth(Breakpoint.MEDIUM - 1);
    });
    it('renders as expected', () => {
      expect(render(<FlowNavigation {...props} onGoBack={jest.fn()} />)).toMatchSnapshot();
    });
  });
});
