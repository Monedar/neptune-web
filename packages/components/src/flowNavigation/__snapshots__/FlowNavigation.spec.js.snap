// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlowNavigation on mobile renders as expected 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="np-header d-flex flex-wrap np-flow-navigation"
      >
        <div
          class="align-items-center d-flex justify-content-start flex__item--6"
        >
          <div
            class="hidden-xs"
          >
            Logo 
            FULL
          </div>
          <div
            class="visible-xs"
          >
             
            <div>
              Logo 
              FLAG
            </div>
          </div>
        </div>
        <div
          class="align-items-center d-flex justify-content-end flex__item--6 "
        >
          <div />
        </div>
        <div
          class="align-items-center d-flex justify-content-center flex__item--12"
        >
          <div
            class="tw-stepper np-flow-navigation__stepper m-t-1"
          >
            <div
              class="progress"
            >
              <div
                class="progress-bar-filler"
                style="width: 0%;"
              />
              <div
                class="progress-bar-ending"
                style="width: 0%;"
              />
            </div>
            <ul
              class="tw-stepper-steps p-t-1 m-b-0"
            >
              <li
                class="
          hidden-xs
          tw-stepper__step
          tw-stepper__step--active
          
        "
                style="left: 0%;"
              >
                <button
                  class="btn-unstyled tw-stepper__step-label"
                  disabled=""
                  type="button"
                >
                  <small>
                    label-0
                  </small>
                </button>
              </li>
              <li
                class="
          hidden-xs
          tw-stepper__step
          
          
        "
                style="left: 100%;"
              >
                <button
                  class="btn-unstyled tw-stepper__step-label"
                  disabled=""
                  type="button"
                >
                  <small>
                    label-1
                  </small>
                </button>
              </li>
            </ul>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="np-header d-flex flex-wrap np-flow-navigation"
    >
      <div
        class="align-items-center d-flex justify-content-start flex__item--6"
      >
        <div
          class="hidden-xs"
        >
          Logo 
          FULL
        </div>
        <div
          class="visible-xs"
        >
           
          <div>
            Logo 
            FLAG
          </div>
        </div>
      </div>
      <div
        class="align-items-center d-flex justify-content-end flex__item--6 "
      >
        <div />
      </div>
      <div
        class="align-items-center d-flex justify-content-center flex__item--12"
      >
        <div
          class="tw-stepper np-flow-navigation__stepper m-t-1"
        >
          <div
            class="progress"
          >
            <div
              class="progress-bar-filler"
              style="width: 0%;"
            />
            <div
              class="progress-bar-ending"
              style="width: 0%;"
            />
          </div>
          <ul
            class="tw-stepper-steps p-t-1 m-b-0"
          >
            <li
              class="
          hidden-xs
          tw-stepper__step
          tw-stepper__step--active
          
        "
              style="left: 0%;"
            >
              <button
                class="btn-unstyled tw-stepper__step-label"
                disabled=""
                type="button"
              >
                <small>
                  label-0
                </small>
              </button>
            </li>
            <li
              class="
          hidden-xs
          tw-stepper__step
          
          
        "
              style="left: 100%;"
            >
              <button
                class="btn-unstyled tw-stepper__step-label"
                disabled=""
                type="button"
              >
                <small>
                  label-1
                </small>
              </button>
            </li>
          </ul>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`FlowNavigation renders as expected 1`] = `
<div>
  <div
    class="np-header d-flex flex-wrap np-flow-navigation"
  >
    <div
      class="align-items-center d-flex justify-content-start"
    >
      <div
        class="hidden-xs"
      >
        Logo 
        FULL
      </div>
      <div
        class="visible-xs"
      >
         
        <div>
          Logo 
          FLAG
        </div>
      </div>
    </div>
    <div
      class="align-items-center d-flex justify-content-end order-2"
    >
      <div />
    </div>
    <div
      class="align-items-center d-flex justify-content-center order-1 flex-grow-1"
    >
      <div
        class="tw-stepper np-flow-navigation__stepper m-t-1"
      >
        <div
          class="progress"
        >
          <div
            class="progress-bar-filler"
            style="width: 0%;"
          />
          <div
            class="progress-bar-ending"
            style="width: 0%;"
          />
        </div>
        <ul
          class="tw-stepper-steps p-t-1 m-b-0"
        >
          <li
            class="
          hidden-xs
          tw-stepper__step
          tw-stepper__step--active
          
        "
            style="left: 0%;"
          >
            <button
              class="btn-unstyled tw-stepper__step-label"
              disabled=""
              type="button"
            >
              <small>
                label-0
              </small>
            </button>
          </li>
          <li
            class="
          hidden-xs
          tw-stepper__step
          
          
        "
            style="left: 100%;"
          >
            <button
              class="btn-unstyled tw-stepper__step-label"
              disabled=""
              type="button"
            >
              <small>
                label-1
              </small>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
